### ISSUE TITLE
Client-side routing breaks layouts

### ISSUE THREAD

**Issue by jaydevelopsstuff** [vikejs/vike-solid#113]
When using client side navigation (e.g. clicking anchor links), Vike fails to add/remove Layouts from the page.

https://github.com/user-attachments/assets/b428acaa-2832-4ec7-834c-76cd61da129e

[MRE Repo](https://github.com/jaydevelopsstuff/vike-solid-layout-mre)
1. Install dependencies w/ pnpm
2. Run `pnpm dev` and open in browser
3. Click the links and observe that the (in this case) nested layout is not applied/removed unless the page is refreshed

**Comment by brillout** [comment]
@Blankeos Maybe there is an issue with the cumulative +Layout implementation?

**Comment by Blankeos** [comment]
Looking into this!

**Comment by Blankeos** [comment]
Hi @jaydevelopsstuff. Looked into your MRE, thanks for this, I could play around with the issue immediately! Found that this isn't a Vike issue. It's is a **very common misuse of SolidJS** if you're coming from React. 

**TL;DR; You're destructuring props. Don't do that. I have some tips below though.** I made a [PR](https://github.com/jaydevelopsstuff/vike-solid-layout-mre/pull/1) on your repo btw for the fix.

### Few tips about SolidJS and props
- You lose reactivity if you destructure props
- Personally, I don't like to destructure, but here's a cheatsheet for most of your usecases:
```tsx
// ❌  bad
import { FlowProps } from "solid-js"
const MyComponent = (props: FlowProps) => {
   const { children } = props;
   // ...
}

// ❌  bad
import { FlowProps } from "solid-js"
const MyComponent = ({ children }: FlowProps) => {
   const { children } = props;
   // ...
}

// ✅  good (also personally useful because you know which ones are props from a glance)
import { FlowProps } from "solid-js"
import { mergeProps } from "solid-js"
const MyComponent = (props: FlowProps) => {
   // ...
}

// ✅ good (if you really want to destructure)
import { mergeProps, FlowProps } from "solid-js"
const MyComponent = (props: FlowProps) => {
   // 1. First way: always use if you want to destructure and also add default values. 
   const  { children } = mergeProps({ age = 14}, props) 

   // 2. Second Way: or still keep it inside a parent object (I like this personally because of the  same reason above)
   const _props = mergeProps({ age = 14}, props)
   
   // ...
}

```

Also some improvements, I could add in your code (this doesn't matter though, I just realized):
- [File Structure](https://vike.dev/file-structure#basic). Your `+Page.tsx` that lives in the `/` route shouldn't be in `src/pages/+Page.tsx` (I guess that's one difference from Next.js and SvelteKit). But apparently, it still works without putting it inside `index/+Page.tsx` (I didn't know that @brillout lol)

<img width="763" alt="image" src="https://github.com/user-attachments/assets/0745d45e-dec5-49bb-ab52-4f21ee25bd7c">

---

By the way if this helped you, I'll just plug my SolidJS + Vike boilerplates here:
- [solid-hop](https://github.com/blankeos/solid-hop) - simpler (like a `create-next` `create-solid` `create-svelte` barebones starter for SolidJS)
- [solid-launch](https://github.com/blankeos/solid-launch) - fully-fledged with auth, folder structures, conventions, snippets, etc.

**Comment by jaydevelopsstuff** [comment]
Thanks so much for the help, just a dumb mistake.

**Comment by Blankeos** [comment]
Happy to help @jaydevelopsstuff. It's perfectly okay to make that mistake. It's just the React-to-Solid jet lag haha. I made the same mistake before lol.

---

Also, hi @brillout, I think I missed reading your reply haha.
 
> do you mean that it should be defined in src/pages/index/+Page.tsx instead? Why is that? 👀

Yup, based on the [file structure docs](https://vike.dev/file-structure#basic), I never really thought `src/pages/+Page.tsx` was possible haha. But after reading it again, yeah it actually is there `src/pages/**/+Page.js`. That's cool! 

Will probably still stick with `src/pages/index/+Page.tsx` though because `+data.ts` or `+config.ts` can be placed there if needed.


> These look great! We could add them to vike-solid's readme/docs if you want.

Really appreciate that coming from you! Sounds awesome! Yeah, I think it would help the Vike + Solid community a lot if it's on the examples.


> Also, if you're up for it, you could co-lead `vike-solid` with us

It would be a really cool opportunity, but not sure if I'm ready for the obligation. What does the co-lead usually do? Feel free to dm me on Discord! 

In any case, I'll do my best answering questions in the community. I think it helps everyone and the framework in the long term. Would really love to see Vike becoming mainstream with giants like NextJS and upcoming ones like Tanstack Router.

**Comment by brillout** [comment]
> It would be a really cool opportunity, but not sure if I'm ready for the obligation. What does the co-lead usually do?

You can contribute however you want and feel like.

Examples:
 - Have a look [at current issues](https://github.com/vikejs/vike-solid/issues). But most are currently already being worked on by @phonzammi (although he'll probably very much welcome some help). One issues that is open and not taken by anyone is https://github.com/vikejs/vike-solid/issues/96, in case that's something that sounds appealing to you. 
 -  Follow new issues (by clicking the watch button at https://github.com/vikejs/vike-solid) while trying to fix them / reply to questions.
 - Implement new Vike extensions for `vike-solid`. See for example https://github.com/vikejs/vike-solid/tree/main/packages/vike-solid-query and https://github.com/vikejs/vike-react/tree/main/packages/vike-react-apollo.
